crates/storage/nippy-jar/src/cursor.rs

```diff
@@ -139,29 +108,10 @@ impl<'a, H: NippyJarHeader> NippyJarCursor<'a, H> {
     /// stored in file.
     pub fn row_by_key_with_cols(
         &mut self,
-        key: &[u8],
-        mask: usize,
+        _key: &[u8],
+        _mask: usize,
     ) -> Result<Option<RefRow<'_>>, NippyJarError> {
-        if let (Some(filter), Some(phf)) = (&self.jar.filter, &self.jar.phf) {
-            // TODO: is it worth to parallelize both?
-
-            // May have false positives
-            if filter.contains(key)? {
-                // May have false positives
-                if let Some(row_index) = phf.get_index(key)? {
-                    self.row = self
-                        .jar
-                        .offsets_index
-                        .access(row_index as usize)
-                        .expect("built from same set") as u64;
-                    return self.next_row_with_cols(mask)
-                }
-            }
-        } else {
-            return Err(NippyJarError::UnsupportedFilterQuery)
-        }
-
-        Ok(None)
+        Err(NippyJarError::UnsupportedFilterQuery)
```

Comment:
why can't we remove this function if this now always returns an error, this is also not a trait function, so unclear why removing this is sound

Reply:
fair, we can remove it completely